corehq/apps/analytics/utils.py

```diff
@@ -174,13 +181,21 @@ def _get_contact_ids_to_delete(list_of_emails, retry_num=0):
             req.raise_for_status()
         except (ConnectionError, requests.exceptions.HTTPError) as e:
             metrics_counter(
-                'commcare.hubspot_data.retry.get_contact_ids_for_emails'
+                'commcare.hubspot_data.retry.get_contact_ids_for_emails',
+                tags={
+                    'emails': list_of_emails.join(', '),
+                }
             )
             if retry_num <= MAX_API_RETRIES:
                 return _get_contact_ids_to_delete(list_of_emails, retry_num + 1)
             else:
-                logger.error(f"Failed to get Hubspot contact ids for emails "
-                             f"{list_of_emails.join(', ')} due to {str(e)}.")
+                metrics_counter(
+                    'commcare.hubspot_data.error.get_contact_ids_for_emails',
+                    tags={
+                        'emails': list_of_emails.join(', '),
+                        'error': str(e),
```

Comment:
Tags should have few distinct values. I believe we have to pay extra for lots of tag combinations.

- Might be worth sending the list of email addresses to a database or log file.
- It's not clear how many distinct values are produced by `str(e)`, but may want to revise if many.

Reply:
ah got it. I think I'm most interested in the error codes vs the email addresses. If it looks like it's getting to be a lot (which I don't think is the case based on the sentry errors), then I'll revise